.github/workflows/pypi.yml

```diff
@@ -176,3 +176,50 @@ jobs:
         with:
           command: upload
           args: --non-interactive --skip-existing wheels-*/*
+
+  docker-build:
+    name: Build and publish Docker image
+    runs-on: ubuntu-latest
+    env:
+      ZIZMOR_IMAGE: ghcr.io/woodruffw/zizmor
+    environment:
+      name: docker
+    if: ${{ startsWith(github.ref, 'refs/tags/') || github.event_name == 'workflow_dispatch' }}
+    # Only run if the release job was successfully completed
+    needs: [release]
```

Comment:
There might be a bit of a race condition with this, sadly: PyPI's propagation can take a few minutes IME, so running immediately after a `release` might fail due to the index not listing the release yet.

I think there are two (easy-ish) options here: 

1. Add some kind of blocking "wait for PyPI" job so that the flow goes `release -> wait-for-pypi -> docker-build` instead
2. Move this to a separate `docker-publish.yml` or similar that can be triggered separately, e.g. by a `workflow_dispatch` with a `tag:` input

I'd be OK with either of these but a slight preference for (2), especially since we can give it a `cron` trigger that'll do the update automatically 🙂

Reply:
I was initially going for 2-ish, but then I realized the trigger would need to be other than a release tag so that this did not end up running _before_ the PyPI release was ready, so I moved this inside the PyPI workflow, though I completely forgot about the potential delays on the PyPI side.

Let me move this out into a separate workflow as in 2) and we can sort out the trigger event (`cron` or something else).

Reply:
Another alternative could be using the built wheels (already uploaded to the artefacts) to install zizmor (thus avoiding any potential delays on the PyPI side of things) so all the workflows would trigger from the same release tag event.

Reply:
That could work, although in that case we'd need to reimplement the wheel/tag matching, right? I think a separate workflow + trigger is good for now 🙂